atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheIntegrationTest.java

```diff
@@ -236,25 +236,23 @@ public void cacheClearedOnSnapshotUpdate() {
     }
 
     @Test
-    public void getEventsForTransactionsReturnsSnapshotWithOldEvents() {
+    public void getEventsForTransactionsReturnsSnapshotWithCondensedEvents() {
```

Comment:
Can you also add tests here? I think there's a bunch of edge cases to cover around the snapshot computation piece? Or add unit tests, whatever works for you.

Reply:
we now have tests for:
1. no condensing
2. some condensing
3. max condensing

I think that covers all the main cases.